pkg/controller/notification/mailgun_notification_delivery_service_test.go

```diff
@@ -0,0 +1,50 @@
+package notification
+
+import (
+	"context"
+	"github.com/codeready-toolchain/host-operator/pkg/templates/notificationtemplates"
+	"github.com/mailgun/mailgun-go/v4"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+type MailgunAPIBaseOption struct {
+	apiBase string
+}
+
+func (o *MailgunAPIBaseOption) ApplyToMailgun(mg mailgun.Mailgun) {
+	mg.SetAPIBase(o.apiBase)
+}
+
+func NewMailguAPIBaseOption(url string) MailgunOption {
+	return &MailgunAPIBaseOption{apiBase: url}
+}
+
+func TestMailgunNotificationDeliveryService(t *testing.T) {
+	// given
+	mgs := mailgun.NewMockServer()
+	opt := NewMailguAPIBaseOption(mgs.URL())
+	mgConfig := NewMockMailgunConfiguration("mg.foo.com", "abcd12345", "noreply@foo.com")
+	notCtx := &NotificationContext{
+		UserID:      "jsmith123",
+		FirstName:   "John",
+		LastName:    "Smith",
+		Email:       "jsmith@redhat.com",
+		CompanyName: "Red Hat",
+	}
+
+	templateLoader := NewMockTemplateLoader(&notificationtemplates.NotificationTemplate{
+		Subject: "foo",
+		Content: "bar",
+		Name:    "test",
+	})
+
+	t.Run("test mailgun notification delivery service send", func(t *testing.T) {
+		// when
+		mgds := NewMailgunNotificationDeliveryService(mgConfig, templateLoader, opt)
+		err := mgds.Send(context.Background(), notCtx, "test")
```

Comment:
it might be worthwhile to also add in tests for the error conditions. for instance if a template is not found.

Reply:
Yes, please add tests for errors too.

Reply:
Good idea, I had the same thought when looking at the coverage report.  By the way, is coverage broken right now?  I'm getting all sorts of stuff included in the coverage report that isn't part of this PR.

Reply:
Yeah... codecov reports for PRs have not been very reliable for some time.

Reply:
I wonder if there is anything in the mailgun server (mock server in this case) we can use to actually check that the api call was made?

Reply:
I think there is (the mock API seems to store events) however it doesn't appear there is any programmatic access to these, instead they provide an events endpoint which can list the events, from this I guess they would need to be parsed and converted into objects before we could write any meaningful tests.  To me this cost/benefit seems like it wouldn't be worth it, just for the sake of confirming that the mailgun API actually does what it's supposed to.

Reply:
It looks like we could actually filter on event types - https://documentation.mailgun.com/en/latest/api-events.html#filter-field. There is the event type `delivered` which we could filter for - https://documentation.mailgun.com/en/latest/api-events.html#event-types